src/artifact_providers/__tests__/base.test.ts

```diff
@@ -9,33 +9,73 @@ describe('parseFilterOptions', () => {
     expect(parsedFilters).not.toHaveProperty('excludeNames');
   });
 
-  test('undefined properties', () => {
+  test.each([
+    [undefined, undefined],
+    [undefined, '/onlyExclude/'],
+    ['/onlyInclude/', undefined],
+    ['/include/', '/exclude/'],
+  ])('undefined and string properties', (includeNames, excludeNames) => {
     const rawFilters: RawFilterOptions = {
-      includeNames: undefined,
-      excludeNames: undefined,
+      includeNames: includeNames,
+      excludeNames: excludeNames,
     };
     const parsedFilters = parseFilterOptions(rawFilters);
-    expect(parsedFilters).not.toHaveProperty('includeNames');
-    expect(parsedFilters).not.toHaveProperty('excludeNames');
+
+    if (includeNames) {
+      expect(parsedFilters.includeNames).toStrictEqual(
+        stringToRegexp(includeNames)
+      );
+    } else {
+      expect(parsedFilters.includeNames).not.toBeDefined();
+    }
+    if (excludeNames) {
+      expect(parsedFilters.excludeNames).toStrictEqual(
+        stringToRegexp(excludeNames)
+      );
+    } else {
+      expect(parsedFilters.excludeNames).not.toBeDefined();
+    }
   });
 
-  test('string properties', () => {
-    const stringFilter = '/testFilter/';
+  test.each([
+    [undefined, undefined],
+    [undefined, new RegExp('onlyExclude')],
+    [new RegExp('onlyInclude'), undefined],
+    [new RegExp('include'), new RegExp('exclude')],
+  ])('undefined and regex properties', (includeNames, excludeNames) => {
     const rawFilters: RawFilterOptions = {
-      includeNames: stringFilter,
+      includeNames: includeNames,
+      excludeNames: excludeNames,
     };
     const parsedFilters = parseFilterOptions(rawFilters);
-    expect(parsedFilters.includeNames).toStrictEqual(
-      stringToRegexp(stringFilter)
-    );
+
+    if (includeNames) {
+      expect(parsedFilters.includeNames).toStrictEqual(includeNames);
+    } else {
+      expect(parsedFilters.includeNames).not.toBeDefined();
+    }
+    if (excludeNames) {
+      expect(parsedFilters.excludeNames).toStrictEqual(excludeNames);
+    } else {
+      expect(parsedFilters.excludeNames).not.toBeDefined();
+    }
   });
 
-  test('regex properties', () => {
-    const regexFilter = stringToRegexp('/testFilter/');
-    const rawFilters: RawFilterOptions = {
-      includeNames: regexFilter,
+  test('string and regex properties', () => {
+    const strIncNames: RawFilterOptions = {
+      includeNames: '/string/',
+      excludeNames: new RegExp('regexp'),
     };
-    const parsedFilters = parseFilterOptions(rawFilters);
-    expect(parsedFilters.includeNames).toStrictEqual(regexFilter);
+    const parsedStrInc = parseFilterOptions(strIncNames);
+    expect(parsedStrInc.includeNames).toStrictEqual(stringToRegexp('/string/'));
```

Comment:
Using `stringToRegexp` for tests is a bit implementation-specific. A better way would be to actually check the filters to match a certain string. It would also make your test case above a lot clearer.